### ISSUE TITLE
Do not remove PURE comment after typed declaration

### ISSUE THREAD

**Issue by LongTengDao** [microsoft/TypeScript#32060]
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:**

**Code**

```ts
const a :Type = /*#__PURE__*/ function () {}();
```

**Expected behavior:**

```js
const a = /*#__PURE__*/ function () {}();
```

**Actual behavior:**

```js
const a = function () {}();
```

**Playground Link:** 

**Related Issues:**

**Comment by AnyhowStep** [comment]
Why not make it a JSDoc?

```ts
/**
 * __PURE__
 */
const a : Type = function () {}();
```

**Comment by AlCalzone** [comment]
This is https://github.com/microsoft/TypeScript/issues/13721 I think

**Comment by LongTengDao** [comment]
@AnyhowStep Thanks for reply. I didn't know that. But only `/*#__PURE__*/` is widely used in downstream tools, like `rollup` (pack tool) or `uglify`/`terser` (minify tool).

BTW: I wonder why the comment removed, it's just same line with types syntax, not same part of an expression, even not same side of assignment equal mark...

**Comment by ashi009** [comment]
@AlCalzone 

> This is #13721 I think

#13721 is mainly for classes, which is fixed in #16631. This is actually #7770.

@LongTengDao 

A workaround is to add parentheses around the `pure` expression, which will preserve the comment until this is fixed.

```ts
const a : Type = (/*#__PURE__*/ function () {}());
// var a = ( /*#__PURE__*/function () { }());
```

**Comment by LongTengDao** [comment]
@ashi009 Oh, that works, thank you!

Anyway, I wrote my personal ts transpiler ([@ ltd/j-ts](https://GitHub.com/LongTengDao/j-ts/)) ;)

**Comment by LongTengDao** [comment]
seems fixed in 5.3

**Comment by jakebailey** [comment]
Yes, this was fixed by #47407 (and #47158 was a dupe of this one).
